logger.go

```diff
@@ -72,87 +84,139 @@ const (
 	sFatal
 )
 
-func output(s severity, txt string) {
-	logLock.Lock()
-	defer logLock.Unlock()
+type logger struct {
+	infoLog  *log.Logger
+	errorLog *log.Logger
+	fatalLog *log.Logger
+	logLock  sync.Mutex
```

Comment:
Optional: Could this sync.Mutex simply be embedded? Especially as logger is a non-exported type.